examples/connection_snippets.rs

```diff
@@ -0,0 +1,38 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+extern crate mongodb;
+
+#[cfg(feature = "tokio-runtime")]
+use tokio::main;
+
+#[cfg(feature = "async-std-runtime")]
+use async_std::main;
+
+use mongodb::error::Result;
+
+#[cfg(feature = "sync")]
+fn main() -> Result<()> {
+    use mongodb::sync::Client;
+    let client = Client::with_uri_str(
+        "mongodb+srv://<username>:<password>@<cluster-url>/<dbname>?w=majority",
+    )?;
+    let database = client.database("test");
+    Ok(())
+}
+
+#[cfg(not(feature = "sync"))]
+#[crate::main]
+async fn main() -> Result<()> {
+    // CONNECTION EXAMPLE STARTS HERE
```

Comment:
Combined the two async mains here. The example still requires there be a sync main so that everything compiles with the sync feature flag, but I added comments to indicate where the example snippet is.